### ISSUE TITLE
Issues with setting up conda - windows 

### ISSUE THREAD

**Issue by whalera1901** [biostars/biostar-handbook-issues#140]
Hi,

I'm following along with the new edition of the handbook and am getting "conda: command not found" after setting everything up as the book says 

I've tried redoing the steps several times, but the results are the same. I also tried typing in "bash miniconda-installer.sh -u" after it said that "my directory already exists" to see if that would fix anything (this was after my first few tries of following the book exactly). After trying out the -u option, it asked if I wanted to initialize it with "init" so I said yes. It then prompted me to close and open the terminal. I then tried going through the steps in the book again and still get the conda : command not found error. 

Here is what my latest run looked like 
![conda error](https://user-images.githubusercontent.com/56034839/108154547-18a33600-70ab-11eb-80bb-9673c991fec8.PNG)

Sorry if anything is unclear as I'm completely new to this kind of stuff and thanks for your help in advance!

**Comment by ialbert** [comment]
make sure to initialize cone there is one extra step described there:

    miniconda3/condabin/conda init
    
then close and re-open the terminal.

**Comment by ialbert** [comment]
you will need to close and reopen the terminal for the changes to take effect

**Comment by whalera1901** [comment]
Hi thanks for replying! I've actually already tried that and then reopened
the terminal and it still has the conda: command not found error.

I've tried typing that in a few times actually after redoing all the steps
with the same result.

**Comment by ialbert** [comment]
This sounds like a path problem, you either are not in bash (perhaps you are running a different shell). What does this print?

    echo $0

on my system it prints:

   -bash

the way programs are found is governed by the so called PATH variable, there is a section in the handbook that explains the PATH. 

https://www.biostarhandbook.com/bashpath.html

You shell is not seeing the conda location for some reason. It is either that the bash shell is wrong, or your have applied some steps incorrectly.

**Comment by whalera1901** [comment]
Typing that in I see -bash
[image: image.png]
Does this mean the bash shell is fine and I applied some steps incorrectly?
If I did the steps incorrectly, should I just go through them all again or
is there something I need to undo since I've already tried going through
them before?

Thank you for the help!

**Comment by whalera1901** [comment]
Hi,

Sorry for another email so soon. This is what I'm getting as I'm trying to
redo the "install conda" steps. Should I still proceed with the next step
despite this result?
[image: image.png]

Caroline Nitirahardjo

**Comment by ialbert** [comment]
please show the text, 

no need to take screenshots, the terminal is text, you can highlight with your mouse and copy /paste it into the text box

**Comment by whalera1901** [comment]
Opps sorry about that! Here is the text;

whalers1845@DESKTOP-IJ0BUL2:~$ URL=https:/
repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
whalers1845@DESKTOP-IJ0BUL2:~$ curl $URL > miniconda-installer.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time
 Current
                                 Dload  Upload   Total   Spent    Left
 Speed
100 89.8M  100 89.8M    0     0  3890k      0  0:00:23  0:00:23 --:--:--
4081k
whalers1845@DESKTOP-IJ0BUL2:~$ bash miniconda-installer.sh -b
ERROR: File or directory already exists: '/home/whalers1845/miniconda3'
If you want to update an existing installation, use the -u option.
whalers1845@DESKTOP-IJ0BUL2:~$ miniconda3/condabin/conda init
no change     /home/whalers1845/miniconda3/condabin/conda
no change     /home/whalers1845/miniconda3/bin/conda
no change     /home/whalers1845/miniconda3/bin/conda-env
no change     /home/whalers1845/miniconda3/bin/activate
no change     /home/whalers1845/miniconda3/bin/deactivate
no change     /home/whalers1845/miniconda3/etc/profile.d/conda.sh
no change     /home/whalers1845/miniconda3/etc/fish/conf.d/conda.fish
no change     /home/whalers1845/miniconda3/shell/condabin/Conda.psm1
no change     /home/whalers1845/miniconda3/shell/condabin/conda-hook.ps1
no change
/home/whalers1845/miniconda3/lib/python3.8/site-packages/xontrib/conda.xsh
no change     /home/whalers1845/miniconda3/etc/profile.d/conda.csh
no change     /home/whalers1845/.bashrc
No action taken.
whalers1845@DESKTOP-IJ0BUL2:~$

**Comment by ialbert** [comment]
I meant to the question I posed, what does:

    echo $0

print, also since were are there what does 

    echo $PATH 

print?

**Comment by whalera1901** [comment]
Here's what prints when I type echo $0 and echo $PATH


whalers1845@DESKTOP-IJ0BUL2:~$ echo $0
-bash
whalers1845@DESKTOP-IJ0BUL2:~$ echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt/c/Program
Files/WindowsApps/CanonicalGroupLimited.UbuntuonWindows_2004.2020.812.0_x64__79rhkp1fndgsc:/mnt/c/Program
Files (x86)/Common Files/Oracle/Java/javapath:/mnt/c/Program
Files/AdoptOpenJDK/jdk-11.0.8.10-hotspot/bin:/mnt/c/Program
Files/AdoptOpenJDK/jre-11.0.7.10-hotspot/bin:/mnt/c/Program
Files/AdoptOpenJDK/jre-8.0.252.09-hotspot/bin:/mnt/c/Program Files
(x86)/AdoptOpenJDK/jre-8.0.252.09-hotspot/bin:/mnt/c/Windows/system32:/mnt/c/Windows:/mnt/c/Windows/System32/Wbem:/mnt/c/Windows/System32/WindowsPowerShell/v1.0/:/mnt/c/Windows/System32/OpenSSH/:/mnt/c/Program
Files/MEGA-X:/mnt/c/Users/Sean/AppData/Local/Microsoft/WindowsApps:/snap/bin
whalers1845@DESKTOP-IJ0BUL2:~$

**Comment by ialbert** [comment]
this is a curious error, for whatever reason conda does not seem to have been added to the PATH. We'll try to track down while your `conda init` did not work/ What does this print:

    cat ~/.bashrc | grep conda -A 5

**Comment by whalera1901** [comment]
Hi sorry for the delay! Here is what I get

whalers1845@DESKTOP-IJ0BUL2:~$ cat ~/.bashrc | grep conda -A 5
# >>> conda initialize >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$('/home/whalers1845/miniconda3/bin/conda' 'shell.bash'
'hook' 2> /dev/null)"
if [ $? -eq 0 ]; then
    eval "$__conda_setup"
else
    if [ -f "/home/whalers1845/miniconda3/etc/profile.d/conda.sh" ]; then
        . "/home/whalers1845/miniconda3/etc/profile.d/conda.sh"
    else
        export PATH="/home/whalers1845/miniconda3/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda initialize <<<

whalers1845@DESKTOP-IJ0BUL2:~$
Caroline Nitirahardjo

**Comment by ialbert** [comment]
everything is set up correctly, it should work

are you sure that you actually exit the application and not just minimize it when you close and reopen it

make sure to exit the whole thing

**Comment by whalera1901** [comment]
I'm fairly certain...I exited and restarted my computer to make sure and
I'm still getting this:

whalers1845@DESKTOP-IJ0BUL2:~$ conda
conda: command not found
whalers1845@DESKTOP-IJ0BUL2:~$


Sorry for the trouble!

**Comment by ialbert** [comment]
my next recommendation would be to reinstall conda:

    # Reverts the changes to bash
    ~/miniconda3/condabin/conda init --reverse
    
    # Removes all of miniconda
    rm -rf miniconda3

close the terminal and reopen
    
    # Rerun conda install
    bash miniconda-installer.sh -b
    
    # Rerun the init
    ~/miniconda3/condabin/conda init
    
now close the terminal and open it again (no need to restart the system)    

let's see if conda can be run now

**Comment by whalera1901** [comment]
Here's what I'm getting:

  whalers1845@DESKTOP-IJ0BUL2:~$ ~/miniconda3/condabin/conda init --reverse
no change     /home/whalers1845/miniconda3/condabin/conda
no change     /home/whalers1845/miniconda3/bin/conda
no change     /home/whalers1845/miniconda3/bin/conda-env
no change     /home/whalers1845/miniconda3/bin/activate
no change     /home/whalers1845/miniconda3/bin/deactivate
no change     /home/whalers1845/miniconda3/etc/profile.d/conda.sh
no change     /home/whalers1845/miniconda3/etc/fish/conf.d/conda.fish
no change     /home/whalers1845/miniconda3/shell/condabin/Conda.psm1
no change     /home/whalers1845/miniconda3/shell/condabin/conda-hook.ps1
no change
/home/whalers1845/miniconda3/lib/python3.8/site-packages/xontrib/conda.xsh
no change     /home/whalers1845/miniconda3/etc/profile.d/conda.csh
modified      /home/whalers1845/.bashrc

==> For changes to take effect, close and re-open your current shell. <==

whalers1845@DESKTOP-IJ0BUL2:~$ rm -rf miniconda3
whalers1845@DESKTOP-IJ0BUL2:~$

*Second Part after re-opening*

whalers1845@DESKTOP-IJ0BUL2:~$ bash miniconda-installer.sh -b
PREFIX=/home/whalers1845/miniconda3
Unpacking payload ...
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/whalers1845/miniconda3

  added / updated specs:
    - _libgcc_mutex==0.1=main
    - brotlipy==0.7.0=py38h27cfd23_1003
    - ca-certificates==2020.10.14=0
    - certifi==2020.6.20=pyhd3eb1b0_3
    - cffi==1.14.3=py38h261ae71_2
    - chardet==3.0.4=py38h06a4308_1003
    - conda-package-handling==1.7.2=py38h03888b9_0
    - conda==4.9.2=py38h06a4308_0
    - cryptography==3.2.1=py38h3c74f83_1
    - idna==2.10=py_0
    - ld_impl_linux-64==2.33.1=h53a641e_7
    - libedit==3.1.20191231=h14c3975_1
    - libffi==3.3=he6710b0_2
    - libgcc-ng==9.1.0=hdf63c60_0
    - libstdcxx-ng==9.1.0=hdf63c60_0
    - ncurses==6.2=he6710b0_1
    - openssl==1.1.1h=h7b6447c_0
    - pip==20.2.4=py38h06a4308_0
    - pycosat==0.6.3=py38h7b6447c_1
    - pycparser==2.20=py_2
    - pyopenssl==19.1.0=pyhd3eb1b0_1
    - pysocks==1.7.1=py38h06a4308_0
    - python==3.8.5=h7579374_1
    - readline==8.0=h7b6447c_0
    - requests==2.24.0=py_0
    - ruamel_yaml==0.15.87=py38h7b6447c_1
    - setuptools==50.3.1=py38h06a4308_1
    - six==1.15.0=py38h06a4308_0
    - sqlite==3.33.0=h62c20be_0
    - tk==8.6.10=hbc83047_0
    - tqdm==4.51.0=pyhd3eb1b0_0
    - urllib3==1.25.11=py_0
    - wheel==0.35.1=pyhd3eb1b0_0
    - xz==5.2.5=h7b6447c_0
    - yaml==0.2.5=h7b6447c_0
    - zlib==1.2.11=h7b6447c_3


The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main
  brotlipy           pkgs/main/linux-64::brotlipy-0.7.0-py38h27cfd23_1003
  ca-certificates    pkgs/main/linux-64::ca-certificates-2020.10.14-0
  certifi            pkgs/main/noarch::certifi-2020.6.20-pyhd3eb1b0_3
  cffi               pkgs/main/linux-64::cffi-1.14.3-py38h261ae71_2
  chardet            pkgs/main/linux-64::chardet-3.0.4-py38h06a4308_1003
  conda              pkgs/main/linux-64::conda-4.9.2-py38h06a4308_0
  conda-package-han~
pkgs/main/linux-64::conda-package-handling-1.7.2-py38h03888b9_0
  cryptography       pkgs/main/linux-64::cryptography-3.2.1-py38h3c74f83_1
  idna               pkgs/main/noarch::idna-2.10-py_0
  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.33.1-h53a641e_7
  libedit            pkgs/main/linux-64::libedit-3.1.20191231-h14c3975_1
  libffi             pkgs/main/linux-64::libffi-3.3-he6710b0_2
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0
  ncurses            pkgs/main/linux-64::ncurses-6.2-he6710b0_1
  openssl            pkgs/main/linux-64::openssl-1.1.1h-h7b6447c_0
  pip                pkgs/main/linux-64::pip-20.2.4-py38h06a4308_0
  pycosat            pkgs/main/linux-64::pycosat-0.6.3-py38h7b6447c_1
  pycparser          pkgs/main/noarch::pycparser-2.20-py_2
  pyopenssl          pkgs/main/noarch::pyopenssl-19.1.0-pyhd3eb1b0_1
  pysocks            pkgs/main/linux-64::pysocks-1.7.1-py38h06a4308_0
  python             pkgs/main/linux-64::python-3.8.5-h7579374_1
  readline           pkgs/main/linux-64::readline-8.0-h7b6447c_0
  requests           pkgs/main/noarch::requests-2.24.0-py_0
  ruamel_yaml        pkgs/main/linux-64::ruamel_yaml-0.15.87-py38h7b6447c_1
  setuptools         pkgs/main/linux-64::setuptools-50.3.1-py38h06a4308_1
  six                pkgs/main/linux-64::six-1.15.0-py38h06a4308_0
  sqlite             pkgs/main/linux-64::sqlite-3.33.0-h62c20be_0
  tk                 pkgs/main/linux-64::tk-8.6.10-hbc83047_0
  tqdm               pkgs/main/noarch::tqdm-4.51.0-pyhd3eb1b0_0
  urllib3            pkgs/main/noarch::urllib3-1.25.11-py_0
  wheel              pkgs/main/noarch::wheel-0.35.1-pyhd3eb1b0_0
  xz                 pkgs/main/linux-64::xz-5.2.5-h7b6447c_0
  yaml               pkgs/main/linux-64::yaml-0.2.5-h7b6447c_0
  zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3


Preparing transaction: done
Executing transaction: done
installation finished.
whalers1845@DESKTOP-IJ0BUL2:~$ ~/miniconda3/condabin/conda init
no change     /home/whalers1845/miniconda3/condabin/conda
no change     /home/whalers1845/miniconda3/bin/conda
no change     /home/whalers1845/miniconda3/bin/conda-env
no change     /home/whalers1845/miniconda3/bin/activate
no change     /home/whalers1845/miniconda3/bin/deactivate
no change     /home/whalers1845/miniconda3/etc/profile.d/conda.sh
no change     /home/whalers1845/miniconda3/etc/fish/conf.d/conda.fish
no change     /home/whalers1845/miniconda3/shell/condabin/Conda.psm1
no change     /home/whalers1845/miniconda3/shell/condabin/conda-hook.ps1
no change
/home/whalers1845/miniconda3/lib/python3.8/site-packages/xontrib/conda.xsh
no change     /home/whalers1845/miniconda3/etc/profile.d/conda.csh
modified      /home/whalers1845/.bashrc

==> For changes to take effect, close and re-open your current shell. <==

*Third Part after re-opening*

whalers1845@DESKTOP-IJ0BUL2:~$ conda
conda: command not found

**Comment by ialbert** [comment]
try running conda with full path and initialize environments that way, does that work?

~/miniconda3/condabin/conda create --name bioinfo python=3.7

the problem is something odd, but also trivial in the end, the program is there just bash does not see it by default

**Comment by whalera1901** [comment]
Here's how it looks. It doesn't look like it's working still to me:

whalers1845@DESKTOP-IJ0BUL2:~$ ~/miniconda3/condabin/conda create --name
bioinfo python=3.7
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/whalers1845/miniconda3/envs/bioinfo

  added / updated specs:
    - python=3.7


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2021.1.19  |       h06a4308_0         121 KB
    certifi-2020.12.5          |   py37h06a4308_0         141 KB
    openssl-1.1.1j             |       h27cfd23_0         2.5 MB
    pip-20.3.3                 |   py37h06a4308_0         1.8 MB
    python-3.7.9               |       h7579374_0        45.3 MB
    readline-8.1               |       h27cfd23_0         362 KB
    setuptools-52.0.0          |   py37h06a4308_0         710 KB
    wheel-0.36.2               |     pyhd3eb1b0_0          33 KB
    ------------------------------------------------------------
                                           Total:        51.0 MB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main
  ca-certificates
 pkgs/main/linux-64::ca-certificates-2021.1.19-h06a4308_0
  certifi            pkgs/main/linux-64::certifi-2020.12.5-py37h06a4308_0
  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.33.1-h53a641e_7
  libedit            pkgs/main/linux-64::libedit-3.1.20191231-h14c3975_1
  libffi             pkgs/main/linux-64::libffi-3.3-he6710b0_2
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0
  ncurses            pkgs/main/linux-64::ncurses-6.2-he6710b0_1
  openssl            pkgs/main/linux-64::openssl-1.1.1j-h27cfd23_0
  pip                pkgs/main/linux-64::pip-20.3.3-py37h06a4308_0
  python             pkgs/main/linux-64::python-3.7.9-h7579374_0
  readline           pkgs/main/linux-64::readline-8.1-h27cfd23_0
  setuptools         pkgs/main/linux-64::setuptools-52.0.0-py37h06a4308_0
  sqlite             pkgs/main/linux-64::sqlite-3.33.0-h62c20be_0
  tk                 pkgs/main/linux-64::tk-8.6.10-hbc83047_0
  wheel              pkgs/main/noarch::wheel-0.36.2-pyhd3eb1b0_0
  xz                 pkgs/main/linux-64::xz-5.2.5-h7b6447c_0
  zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3


Proceed ([y]/n)? y


Downloading and Extracting Packages
openssl-1.1.1j       | 2.5 MB    |
###################################################### | 100%
certifi-2020.12.5    | 141 KB    |
###################################################### | 100%
wheel-0.36.2         | 33 KB     |
###################################################### | 100%
setuptools-52.0.0    | 710 KB    |
###################################################### | 100%
pip-20.3.3           | 1.8 MB    |
###################################################### | 100%
python-3.7.9         | 45.3 MB   |
###################################################### | 100%
ca-certificates-2021 | 121 KB    |
###################################################### | 100%
readline-8.1         | 362 KB    |
###################################################### | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate bioinfo
#
# To deactivate an active environment, use
#
#     $ conda deactivate

whalers1845@DESKTOP-IJ0BUL2:~$ conda activate bioinfo
conda: command not found
whalers1845@DESKTOP-IJ0BUL2:~$ conda
conda: command not found
whalers1845@DESKTOP-IJ0BUL2:~$

**Comment by ialbert** [comment]
use `~/miniconda3/condabin/conda` every time the instuctions say `conda`

in this case it would be:

    ~/miniconda3/condabin/conda activate bioinfo

**Comment by whalera1901** [comment]
Hi,

I tried out what you typed but it still didn't work. However, I redid
everything on a new laptop and it's working fine now. The things I did
differently were: not change the display colours, skip the "make the prompt
nicer" section, and ran "curl $URL > miniconda-installer.sh"  a few times
since it kept cancelling the process without downloading everything (did
this until it reached 100%). Thanks for all your help!

Best,
Caroline Nitirahardjo
